tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTargetedFailPointTest.cs

```diff
@@ -14,29 +14,51 @@
 */
 
 using System.Collections.Generic;
+using System.Net;
+using System.Threading;
 using System.Threading.Tasks;
 using FluentAssertions;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
 using MongoDB.Driver.Core.Servers;
 
 namespace MongoDB.Driver.Tests.JsonDrivenTests
 {
     public sealed class JsonDrivenTargetedFailPointTest : JsonDrivenConfigureFailPointTest
     {
         public JsonDrivenTargetedFailPointTest(IJsonDrivenTestRunner testRunner, Dictionary<string, object> objectMap)
-            : base(testRunner, client: null, objectMap)
+            : base(testRunner, objectMap)
         {
         }
 
         protected override IServer GetServer()
         {
-            var pinnedServer = GetPinnedServer();
-            pinnedServer.Should().NotBeNull();
-            return pinnedServer;
+            var pinnedServerEndpoint = AssertAndGetPinnedServerEndpoint();
```

Comment:
Just call `GetPinnedServer` and don't bother to assert? (In any case the constructor of `EndPointServerSelector` already asserts the same thing).